From ff7bc42069a76bafaaba9bbd73c0d8623475130e Mon Sep 17 00:00:00 2001
From: Marco Felsch <m.felsch@pengutronix.de>
Date: Wed, 30 Oct 2019 18:04:55 +0100
Subject: [PATCH] of: of_path: fix return in case of EPROBE_DEFER

As said in commit 82eb3dff10 ("of_path: handle no driver for device")
this case happens if the driver isn't probed yet. So we should return
-EPROBE_DEFER to signal that.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 drivers/of/of_path.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/of_path.c b/drivers/of/of_path.c
index f8bbf2c..5c3a020 100644
--- a/drivers/of/of_path.c
+++ b/drivers/of/of_path.c
@@ -83,7 +83,7 @@ static int __of_find_path(struct device_node *node, const char *part, char **out
 	}
 
 	if (dev->bus && !dev->driver)
-		return -ENODEV;
+		return -EPROBE_DEFER;
 
 	device_detect(dev);
 
-- 
2.7.4

